### ISSUE TITLE
Possibly minor bug with target for RaLS fake being -1

### ISSUE THREAD

**Issue by samedii** [AlexiaJM/relativistic-f-divergences#8]
I doubt this affects any of your results but is there a motivation for the for the second term in RaLS looking like below?

```
errD = (
    torch.mean((y_pred - torch.mean(y_pred_fake) - y) ** 2)
    + torch.mean((y_pred_fake - torch.mean(y_pred) + y) ** 2)
)
```

I would have expected something like this:

```
def relative_average_mse_loss(prediction_true, prediction_false):
    return (
        F.mse_loss(
            prediction_true - torch.mean(prediction_false),
            torch.ones_like(prediction_true),
        )
        + F.mse_loss(
            prediction_false - torch.mean(prediction_true),
            torch.zeros_like(prediction_true),
        )
    ) / 2
```

You code is equivalent to changing the target to `-1` for the second term. Having `-1` as target is inconsistent with your LSGAN loss `errD_fake = torch.mean((y_pred_fake) ** 2)` where you are using `0`

Like I said, I doubt this really affects anything but I'm wondering what you think is correct

**Comment by AlexiaJM** [comment]
I doubt changing the value does anything, but for classification the +1 / -1 labels makes more sense theoretically. 

In the paper I only proved what I called symmetric cases (assuming E[f(C(x))] + E[f(-C(G(z)))] ). I did this for simplicity but it turns out that what I called symmetric loss functions are also called margin based loss functions (of the form f(yC(x)) where y is the label +1 or -1). See https://en.wikipedia.org/wiki/Loss_functions_for_classification for more info. You'll see that the Square loss in the wiki is that least squares with  +1/-1.

A really cool thing about margin-losses is that using a gradient penalty when using them means that your critic acts like a maximum margin classifier (like SVMs, see https://arxiv.org/pdf/1910.06922.pdf).

I kept LSGAN with 0 label only for consistency with the original paper.
